### PR TITLE
Move save_hyperparameters to its own function

### PR DESCRIPTION
**Pull request by tchaton**
### Motivation

Would like to add `save_hyperparemeters` to Flash Preprocess / Postprocess, so we don't pickle the object.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -1647,36 +1647,7 @@ class ``__init__`` to be ignored
         """
         if not frame:
             frame = inspect.currentframe().f_back
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:1649]: This is in `save_hyperparameters`

```suggestion
```

File: pytorch_lightning/utilities/parsing.py [hunk 2]
```diff
@@ -161,6 +163,106 @@ def flatten_dict(source, result=None):
...
+        ...         super().__init__()
+        ...         # pass argument(s) to ignore as a string or in a list
+        ...         obj.save_hyperparameters(ignore='arg2')
+        ...     def forward(obj, *args, **kwargs):
+        ...         ...
+        >>> model = ManuallyArgsModel(1, 'abc', 3.14)
+        >>> model.hparams
+        "arg1": 1
+        "arg3": 3.14
+    """
```
Comment (carmocca) [inline pytorch_lightning/utilities/parsing.py:231]: Let's just keep the docstring in one place?

```suggestion
    """
    See :meth:`~pytorch_lightning.core.lightning.LightningModule.save:hyperparameters`
    """
```

**Review by carmocca — COMMENTED** [review]
What motivates this change? Just reorganizing?

**Comment by tchaton** [comment]
> What motivates this change? Just reorganizing?

Hey @carmocca, I would like to add `save_hyperparemeters` to Flash `Preprocess / Postprocess`, so we don't pickle the object.

Best,
T.C

File: pytorch_lightning/utilities/parsing.py [hunk 3]
```diff
@@ -161,6 +162,47 @@ def flatten_dict(source, result=None):
     return result
 
 
+def save_hyperparameters(
+    obj: Any,
+    *args,
+    ignore: Optional[Union[Sequence[str], str]] = None,
+    frame: Optional[types.FrameType] = None
+) -> None:
+    """See :meth:`~pytorch_lightning.core.lightning.LightningModule.save_hyperparameters`"""
```
Comment (kaushikb11) [inline pytorch_lightning/utilities/parsing.py:171]: Please change the reference.

Comment (tchaton) [reply pytorch_lightning/utilities/parsing.py:171]: It is fine. If readers get there, they can re-direct themselves to the LightningModule.

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/core/lightning.py [hunk 4]
```diff
@@ -1645,38 +1645,10 @@ class ``__init__`` to be ignored
             "arg1": 1
             "arg3": 3.14
         """
+        # the frame needs to be create in this file.
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:1648]:
```suggestion
        # the frame needs to be created in this file.
```

**Review by awaelchli — APPROVED** [review]

**Pull request merged**